projects/vdk-core/src/vdk/internal/builtin_plugins/job_secrets/exception.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Optional
+
+from vdk.internal.core.errors import BaseVdkError
+from vdk.internal.core.errors import ResolvableBy
+
+
+class SecretsException(BaseVdkError):
+    """
+    Base Exception for all custom exceptions related to the properties.
+    """
+
+    def __init__(
+        self,
+        message: Optional[str] = None,
+        resolvable_by: Optional[ResolvableBy] = None,
+    ):
+        super().__init__(None, resolvable_by, message)
```

Comment:
Why are we passing None for resolvable_by_actual here?

Reply:
resolvable_by_actual is always calculated from resolvable_by. 

I will remove it as a argument to an exception altogether, user should not really be able to set it directly.